OpenRA.Mods.Common/Traits/BotModules/CaptureManagerBotModule.cs

```diff
@@ -140,52 +140,44 @@ void QueueCaptureOrders(IBot bot)
 				: GetActorsThatCanBeOrderedByPlayer(randPlayer);
 
 			var capturableTargetOptions = targetOptions
-				.Select(a => new CaptureTarget<CapturableInfo>(a, "CaptureActor"))
 				.Where(target =>
 				{
-					if (target.Info == null)
-						return false;
-
-					var captureManager = target.Actor.TraitOrDefault<CaptureManager>();
+					var captureManager = target.TraitOrDefault<CaptureManager>();
 					if (captureManager == null)
 						return false;
 
-					return capturers.Any(tp => captureManager.CanBeTargetedBy(target.Actor, tp.Actor, tp.Trait));
+					return capturers.Any(tp => captureManager.CanBeTargetedBy(target, tp.Actor, tp.Trait));
 				})
-				.OrderByDescending(target => target.Actor.GetSellValue())
+				.OrderByDescending(target => target.GetSellValue())
 				.Take(maximumCaptureTargetOptions);
 
 			if (Info.CapturableActorTypes.Any())
-				capturableTargetOptions = capturableTargetOptions.Where(target => Info.CapturableActorTypes.Contains(target.Actor.Info.Name.ToLowerInvariant()));
+				capturableTargetOptions = capturableTargetOptions.Where(target => Info.CapturableActorTypes.Contains(target.Info.Name.ToLowerInvariant()));
 
 			if (!capturableTargetOptions.Any())
 				return;
 
 			var capturesCapturers = capturers.Where(tp => tp.Actor.Info.HasTraitInfo<CapturesInfo>());
 			foreach (var tp in capturesCapturers)
-				QueueCaptureOrderFor(bot, tp.Actor, GetCapturerTargetClosestToOrDefault(tp.Actor, capturableTargetOptions));
+				QueueCaptureOrder(bot, tp.Actor, GetCapturerTargetClosestToOrDefault(tp.Actor, capturableTargetOptions));
 		}
 
-		void QueueCaptureOrderFor<TTargetType>(IBot bot, Actor capturer, CaptureTarget<TTargetType> target) where TTargetType : class, ITraitInfoInterface
+		void QueueCaptureOrder(IBot bot, Actor capturer, Actor target)
 		{
 			if (capturer == null)
 				return;
 
 			if (target == null)
 				return;
 
-			if (target.Actor == null)
-				return;
-
-			bot.QueueOrder(new Order(target.OrderString, capturer, Target.FromActor(target.Actor), true));
-			AIUtils.BotDebug("AI ({0}): Ordered {1} to capture {2}", player.ClientIndex, capturer, target.Actor);
+			bot.QueueOrder(new Order("CaptureActor", capturer, Target.FromActor(target), true));
```

Comment:
`capturer` can't be null, so IMO it would be better to inline this and the `target` null check directly in the one place this is called.